### PR TITLE
Test executor sortedsets

### PR DESCRIPTION
**Pull request by hcwilhelm**
SortedSets API support for the TestExecutor backed by a Map[Element, Score]
I found it easier to have a Map of Element -> Score as the underlying data structure than for example a scala TreeSet.

The implementation is not complete at this point but I wanted to create a PR anyways to get some feedback. 
I am struggling with implementing full SortedSet API support because some commands support an enormous amount of 
Options and also for some commands It is not directly obvious how they should work.

### CODE REVIEW

File: redis/src/main/scala/zio/redis/TestExecutor.scala [hunk 1]
```diff
@@ -972,6 +980,972 @@ private[redis] final class TestExecutor private (
...
+//            newSet         = ss.drop(1)
+//            _             <- sortedSets.put(sk, newSet)
+//          } yield Replies.array(Chunk(sk, min.member, min.score.toString)))
+//            .foldM(_ => STM.retry, result => STM.succeed(result))
+//        )
+//
+      case api.SortedSets.ZAdd =>
+        val key = input(0).asString
+
+        val updateOption = input.map(_.asString).find {
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/TestExecutor.scala:1018]: Does it need conversion to string? Also, `|` might make it cleaner:

```scala
case "XX" | "NX" | "LT" | "GT" => true
case _ => false
```

File: redis/src/main/scala/zio/redis/TestExecutor.scala [hunk 2]
```diff
@@ -972,6 +980,972 @@ private[redis] final class TestExecutor private (
...
+        }
+
+        val incrOption = input.map(_.asString).find {
+          case "INCR" => true
+          case _      => false
+        }
+
+        val optionsCount = updateOption.map(_ => 1).getOrElse(0) + changedOption.map(_ => 1).getOrElse(0) + incrOption
+          .map(_ => 1)
+          .getOrElse(0)
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/TestExecutor.scala:1038]: Please use `fold` instead of `map` + `getOrElse`. This remark applies generally :).

**Review by mijicd — COMMENTED** [review]
I'll make another, more detailed pass as I need to review this using editor to understand everything, but in general, mapping doesn't seem to be troublesome.

File: redis/src/test/scala/zio/redis/ApiSpec.scala [hunk 3]
```diff
@@ -32,13 +32,14 @@ object ApiSpec
...
-        connectionSuite,
-        setsSuite,
-        hyperLogLogSuite,
-        listSuite,
-        hashSuite
+        connectionSuite @@ TestAspect.ignore,
+        setsSuite @@ TestAspect.ignore,
+        hyperLogLogSuite @@ TestAspect.ignore,
+        listSuite @@ TestAspect.ignore,
+        hashSuite @@ TestAspect.ignore,
```
[the remark this answers is not in the document]
Comment (hcwilhelm) [reply redis/src/test/scala/zio/redis/ApiSpec.scala:41]: I ignored them only to run the tests for the sorted set API. That helped me to unclutter the whole test output. I will enable them of course when this PR is in the final state.

**Comment by hcwilhelm** [comment]
> I'll make another, more detailed pass as I need to review this using editor to understand everything, but in general, mapping doesn't seem to be troublesome.

Ok, that's good. I find it way easier to work with the map where I have access to both the value and the score. And so far I could satisfy all the tests I allowed to run ;) And yes I fully understand this is a giant PR and the sorted set Implementation has by far the most lines of code, although I am sure there is a lost of duplication.

File: redis/src/main/scala/zio/redis/TestExecutor.scala [hunk 4]
```diff
@@ -2087,11 +2088,60 @@ private[redis] final class TestExecutor private (
...
+        val key              = input(0).asString
+        val maybeCount        = input.tail.headOption.map(b => b.asString.toLong)
+        val withScoresOption = input.map(_.asString).find(_ == "WITHSCORES")
+
+        orWrongType(isSortedSet(key))(
+          for {
+            scoreMap <- sortedSets.getOrElse(key, Map.empty)
+            asVector = scoreMap.toVector
+            res <- maybeCount match {
+              case None => selectOne(asVector, randomPick).map {
```
Comment (jxnu-liguobin) [inline redis/src/main/scala/zio/redis/TestExecutor.scala:2106]: Is `selectOne` method provided by ZIO?

Comment (hcwilhelm) [reply redis/src/main/scala/zio/redis/TestExecutor.scala:2106]: @jxnu-liguobin no it's defined in the same file on line 2229

**Review by mijicd — APPROVED** [review]

**Pull request merged**